jaeger-spark-dependencies/src/main/java/io/jaegertracing/spark/dependencies/DependenciesSparkJob.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright 2017 The Jaeger Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.jaegertracing.spark.dependencies;
+
+import io.jaegertracing.spark.dependencies.cassandra.CassandraDependenciesJob;
+import io.jaegertracing.spark.dependencies.elastic.ElasticsearchDependenciesJob;
+import java.io.UnsupportedEncodingException;
+import java.net.URL;
+import java.net.URLDecoder;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.TimeZone;
+
+public final class DependenciesSparkJob {
+
+  public static void main(String[] args) throws UnsupportedEncodingException {
+    if (args.length != 1) {
+      throw new IllegalArgumentException("");
+    }
+
+    String storage = System.getenv("STORAGE");
+    if (storage == null) {
+      throw new IllegalArgumentException("Missing environmental variable STORAGE");
+    }
+
+    run(storage, args.length == 1 ? parseDay(args[0]) : System.currentTimeMillis());
+  }
+
+  private static void run(String storage, long day) throws UnsupportedEncodingException {
+    String jarPath = pathToUberJar();
+    if ("elasticsearch".equalsIgnoreCase(storage)) {
+      ElasticsearchDependenciesJob.builder()
+          .jars(jarPath)
+          .day(day)
+          .build()
+          .run();
+    } else if ("cassandra".equalsIgnoreCase(storage)) {
+      CassandraDependenciesJob.builder()
+          .jars(jarPath)
+          .day(day)
+          .build()
+          .run();
+    } else {
+      throw new IllegalArgumentException("Unsupported storage: " + storage);
+    }
+  }
+
+  static String pathToUberJar() throws UnsupportedEncodingException {
```

Comment:
Does this need to be `UberJar`? Could it be `JaegerJar`?

Reply:
I think it's more uber jar, no?

Reply:
May be I should have asked what is the Uber jar?

Reply:
a fat jar with all dependencies.

Reply:
Ah ok - overloaded use of the term `uber` :)

Reply:
ROLF now I get it why did you ask... :laughing: